### PR TITLE
chore: update to improve performance using v8 rather than babel for code coverage

### PR DESCRIPTION
**Pull request by Ibrahimmaga**
# Description
Indicates which provider should be used to instrument code for coverage. Allowed values are babel (default) or v8.

Note that using v8 is considered experimental. This uses V8's builtin code coverage rather than one based on Babel. It is not as well tested, and it has also improved in the last few releases of Node. Using the latest versions of node (v14 at the time of this writing) will yield better results.

 "testURL": "http://localhost",
    "transform": {
      "^.+\\.ts?$": "ts-jest"
      "^.+\\.js?$": "babel-jest" 
      we have add GITHUB_TOKEN to enable the code coverage to code climate
## Motivation & context
Here we are looking for performance.  from local host machine it has decreased total time from 105s to 75s. We will test in CI-workflow to get a better picture.
 

Performance

https://github.com/microsoft/fast/issues/3042
## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [x] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [x] This change causes current functionality to break.

This can affect the testing with Jest and Babel
**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

- [ ] I have added a new component
- [ ] I have modified an existing component
- [x] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,6 +48,7 @@
     "prepare": "lerna run prepare",
     "test": "yarn eslint \"/**/*.{ts}\"",
     "unit-tests": "jest --maxWorkers=4",
+    "coverage": "npx jest --coverage",
```
Comment (awentzel) [inline package.json:51]: We're not using `npx` anywhere so I think we should stick with yarn to be consistent.

File: package.json [hunk 2]
```diff
@@ -83,7 +84,7 @@
     "testURL": "http://localhost",
     "transform": {
       "^.+\\.ts?$": "ts-jest",
-      "^.+\\.js?$": "babel-jest"
+      "^.+\\.js?$": "babel-jest"
```
Comment (awentzel) [inline package.json:87]: I don't see any different with these lines.

File: .github/workflows/ci-weekly.yml [hunk 3]
```diff
@@ -120,24 +120,16 @@ jobs:
         yarn test-firefox:pw 
         yarn test-webkit:pw
 
+    - name: Test Coverage
+      run: coverage
```
Comment (awentzel) [inline .github/workflows/ci-weekly.yml:124]: Does 'coverage' run all of our packages?

**Review by awentzel — CHANGES_REQUESTED** [review]
25% improvement is excellent.  I would like to have a walk-through on this PR to see it in action.  Please schedule a half-hour for me to review live.  Thanks

File: .github/workflows/ci-weekly.yml (hunk 3 above)
Comment (Ibrahimmaga) [reply .github/workflows/ci-weekly.yml]: Yes. It does now

**Review by awentzel — APPROVED** [review]

**Review by awentzel — APPROVED** [review]

File: .github/workflows/ci-nightly.yml [hunk 4]
```diff
@@ -35,8 +35,8 @@ jobs:
       run: lerna run prettier:diff
 
     - name: Testing unit tests
-      run: lerna run test --stream
-
+      run: lerna run test --stream   
+
```
Comment (chrisdholt) [inline .github/workflows/ci-nightly.yml:39]: Should these added spaces be removed?

Comment (awentzel) [reply .github/workflows/ci-nightly.yml:39]: @chrisdholt this space should have been there before. It creates spacing between tasks which is consistent across all yaml pipelines.  So this is accurate.

Comment (chrisdholt) [reply .github/workflows/ci-nightly.yml:39]: @awentzel I think an empty return is expected but I'm not sure about the (several) empty spaces after the text and on the next line. I won't block on it but the above is different than an empty return. Not sure if it'll cause issues or not though.

Comment (Ibrahimmaga) [reply .github/workflows/ci-nightly.yml:39]: This won't cause any issues.

**Review by EisenbergEffect — APPROVED** [review]
Let's add a follow-up task to have GitHub run a coverage check any time a PR is submitted so that we can see whether individual PRs increase or reduce coverage. Or is that already in place? I see it in this PR but not others. Was it this PR that added that?

**Comment by Ibrahimmaga** [comment]
Yes.  it is included in this PR. I have made the changed yesterday. For each PR, we will have a code coverage upload to code climate.

**Comment by EisenbergEffect** [comment]
Excellent. Thank you!

**Review by chrisdholt — APPROVED** [review]

**Comment by awentzel** [comment]
> Let's add a follow-up task to have GitHub run a coverage check any time a PR is submitted so that we can see whether individual PRs increase or reduce coverage. Or is that already in place? I see it in this PR but not others. Was it this PR that added that?

We refactored to ensure code coverage would happen on push to master and pull_request events to solve your concern.

**Pull request merged**